### DISCUSSION TITLE
Support for  Pico-ResTouch-LCD-2.8

### DISCUSSION THREAD

**Discussion by masutokw** [Bodmer/TFT_eSPI#1558] [Compatible displays and setup files]
Hello I couldn´t find a suitable user setup for for this device from Waveshare.
[https://www.waveshare.com/wiki/Pico-ResTouch-LCD-2.8](url)
I have modified other setup from another restouch device and it seems to works fine.
````
// See SetupX_Template.h for all options available

//#define RPI_DISPLAY_TYPE
#define ST7789_DRIVER
// #define RP2040_PIO_SPI // [Edit by Bodmer] Touch not supported when this is used (with TFT_eSPI v2.4.21 onwards)

#define TFT_BACKLIGHT_ON HIGH

#define TFT_MISO 12
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_BL  13
#define TFT_CS   9 // Chip select control pin
#define TFT_DC    8  // Data Command control pin
#define TFT_RST   15  // Reset pin (could connect to RST pin)
//#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

#define TOUCH_CS 16     // Chip select pin (T_CS) of touch screen


#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define TFT_SPI_PORT 1
#define SPI_FREQUENCY  40000000

#define SPI_TOUCH_FREQUENCY  2500000

````

**Comment by Bodmer** [comment]
Thanks. You don't need to define the SPI port when using PIO.

**Reply by masutokw** [reply]
Ichecked and It won´t work if it´s not defined,
Touch  is reporting weird random values in some way related to pressed zone but unusable.
If RPI_DISPLAY_TYPE is defined instead  touch give consistent values but display dont work properly only deformed pixel on upper left corner in portrait orientation.
With waveshare´s and RPI PICO sdk sample code all works but is clearly not  fancy as your implementation

**Comment by Bodmer** [comment]
I am guessing you are using a version of TFT_eSPI that does not have the PIO capability because I would not expect touch to work when PIO is used. At the moment the PIO capability built into later libraries is transmit only.

Which version of TFT_eSPI are you using?

**Reply by masutokw** [reply]
I have got installed The latest available  in Arduino IDE´s library management :  2.4.2

**Reply by Bodmer** [reply]
That explains it. PIO SPI was introduced at 2.4.21 but you seem to have used: #define RP2040_PIO_SPI

That will do nothing in version 2.4.2. In later versions it will disable the SPI reads so touch support is lost. I suggest you remove that line. I will edit your post to avoid confusing others.

**Comment by masutokw** [comment]
After update and comenting out  RP2040_PIO_SPI    the screen works and touch pad works fine too  with Earlephilhower´s  arduino core but using Mbed Os RP2040 still return unusable raw values.
Thank you

**Reply by Bodmer** [reply]
Great good to hear it is working.

Earle's Arduino package was the first available so support did focus on that board package initially.  Later versions of the TFT_eSPI library (v2.4.43 tested) work OK with the Mbed Mbed OS RP2040 and touch. PIO is not supported for the Mbed package though.

**Comment by MYCAMEL222** [comment]
I keep getting the following error when trying to run any of the examples on the Waveshare 2.8 w/ pico. Any ideas what I'm missing....?

Arduino: 1.8.8 (Windows 10), Board: "Raspberry Pi Pico, 2MB (Sketch: 1984KB, FS: 64KB), 125 MHz, Small (-Os) (standard), Enabled, Disabled, None, Pico SDK"

In file included from C:\Users\Mike\Documents\Arduino\libraries\TFT_eSPI\TFT_eSPI.cpp:26:
C:\Users\Mike\Documents\Arduino\libraries\TFT_eSPI\Processors/TFT_eSPI_RP2040.c:18:62: error: 'TFT_SCLK' was not declared in this scope
   18 |     SPIClassRP2040 spi = SPIClassRP2040(SPI_X, TFT_MISO, -1, TFT_SCLK, TFT_MOSI);
      |                                                              ^~~~~~~~
C:\Users\Mike\Documents\Arduino\libraries\TFT_eSPI\Processors/TFT_eSPI_RP2040.c:18:72: error: 'TFT_MOSI' was not declared in this scope; did you mean 'TFT_MISO'?
   18 |     SPIClassRP2040 spi = SPIClassRP2040(SPI_X, TFT_MISO, -1, TFT_SCLK, TFT_MOSI);
      |                                                                        ^~~~~~~~
      |                                                                        TFT_MISO
exit status 1
Error compiling for board Raspberry Pi Pico.

**Comment by masutokw** [comment]
Check your includes 
This is my final working User_Setup.h header file for Pico-Restouch 2.8.
````
//See SetupX_Template.h for all options available

//#define RPI_DISPLAY_TYPE
#define ST7789_DRIVER
//#define ST7789_2_DRIVER 
//#define RP2040_PIO_SPI
#define TFT_WIDTH 240
#define TFT_HEIGHT 320
#define TFT_BACKLIGHT_ON HIGH

#define TFT_MISO 12
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_BL  13
#define TFT_CS   9 // Chip select control pin
#define TFT_DC    8  // Data Command control pin
#define TFT_RST   15  // Reset pin (could connect to RST pin)
//#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

#define TOUCH_CS 16     // Chip select pin (T_CS) of touch screen


#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
// #define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue
#define TFT_RGB_ORDER TFT_BGR  // Colour order Blue-Green-Red
#define SMOOTH_FONT

#define TFT_SPI_PORT 1
#define SPI_FREQUENCY  40000000

#define SPI_TOUCH_FREQUENCY  2500000
#define SUPPORT_TRANSACTIONS
#define SPI_HAS_TRANSACTION

**Reply by MYCAMEL222** [reply]
Did you get the touch portion working using the tft_espi library or did you use another library.

**Reply by masutokw** [reply]
Touch screen works only EarlesPhilPower core when using tft_espi, Mbed doesn´t .So I´m using the first one.

**Comment by MYCAMEL222** [comment]
Thanks for the reply....Took me a little while to figure out I had to define everything in the setup.h file. Got it working and hooked up to 2 i2c devices and works like a charm. Now trying to get the graphics and fonts sorted out.
